sdks/java/core/src/main/java/org/apache/beam/sdk/io/WriteFilesResult.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.io;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PInput;
+import org.apache.beam.sdk.values.POutput;
+import org.apache.beam.sdk.values.PValue;
+import org.apache.beam.sdk.values.TupleTag;
+
+/** The result of a {@link WriteFiles} transform. */
+public class WriteFilesResult implements POutput {
```

Comment:
* It should probably be destination-dependent, i.e. `PCollection<KV<DestinationT, String>>`
* Not a fan of the POutput boilerplate: why do we need a WriteFilesResult in general, why not just return a PCollection - is that because you want to make it forward-compatible?

Reply:
This is what we did in BigQueryIO for exactly that reason - it is forward compatible, and I can imagine wanting to return additional metadata in the future, as well as other things (e.g. dead-letter type accessors like in BigQueryIO).

Added a second getter to return KV<DestinationT, String>(simple use cases of sinks do not set a custom destination type, so want there to be a simple accessor)